opm/models/io/basevanguard.hh

```diff
@@ -110,6 +110,17 @@ public:
         updateGridView_();
     }
 
+    /*!
+     * \brief Add LGRs to the grid, if any. Only supported for CpGrid - for now.
+     */
+    void addLgrs()
+    {
+        if(&asImp_() != this) {
```

Comment:
Please explain the if statement to me. When is it true? Why do we need it?

Reply:
I added a comment